homeassistant/components/intellifire/__init__.py

```diff
@@ -32,79 +44,129 @@
 ]
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up IntelliFire from a config entry."""
-    LOGGER.debug("Setting up config entry: %s", entry.unique_id)
+def _construct_common_data(entry: ConfigEntry) -> IntelliFireCommonFireplaceData:
+    """Convert config entry data into IntelliFireCommonFireplaceData."""
 
-    if CONF_USERNAME not in entry.data:
-        LOGGER.debug("Old config entry format detected: %s", entry.unique_id)
-        raise ConfigEntryAuthFailed
-
-    ift_control = IntellifireControlAsync(
-        fireplace_ip=entry.data[CONF_HOST],
+    return IntelliFireCommonFireplaceData(
+        auth_cookie=entry.data[CONF_AUTH_COOKIE],
+        user_id=entry.data[CONF_USER_ID],
+        web_client_id=entry.data[CONF_WEB_CLIENT_ID],
+        serial=entry.data[CONF_SERIAL],
+        api_key=entry.data[CONF_API_KEY],
+        ip_address=entry.data[CONF_IP_ADDRESS],
+        read_mode=entry.options[CONF_READ_MODE],
+        control_mode=entry.options[CONF_CONTROL_MODE],
     )
-    try:
-        await ift_control.login(
-            username=entry.data[CONF_USERNAME],
-            password=entry.data[CONF_PASSWORD],
-        )
-    except (ConnectionError, ClientConnectionError) as err:
-        raise ConfigEntryNotReady from err
-    except LoginException as err:
-        raise ConfigEntryAuthFailed(err) from err
 
-    finally:
-        await ift_control.close()
 
-    # Extract API Key and User_ID from ift_control
-    # Eventually this will migrate to using IntellifireAPICloud
+async def _async_pseudo_migrate_entry(
```

Comment:
Why is this pseudo? Can we do a minor version refactor?

Reply:
We had a long discussion about this I think earlier. We can't actually do a minor migration because there is no backwards compatibility once the migration has happened.

I believe we started with a proper migration - then - it was suggested to go this direction - then minor migrations came out - and at this point I dont care what to do but I think minor is invalid because there is no backwards compatibility

Reply:
But if we revert to the version before this PR with the same config, does it work?

Reply:
I'm about 95% sure I started with an actual migration and at the time some guy who was probably you - told me not to migrate - just to update the config... and then minor migrations came out and now I'm not sure what to do :) 

Old config:

```json
        data={
            CONF_IP_ADDRESS: "192.168.2.108",
            CONF_USERNAME: "grumpypanda@china.cn",
            CONF_PASSWORD: "you-stole-my-pandas",
            CONF_SERIAL: "3FB284769E4736F30C8973A7ED358123",
            CONF_WEB_CLIENT_ID: "FA2B1C3045601234D0AE17D72F8E975",
            CONF_API_KEY: "B5C4DA27AAEF31D1FB21AFF9BFA6BCD2",
            CONF_AUTH_COOKIE: "B984F21A6378560019F8A1CDE41B6782",
            CONF_USER_ID: "52C3F9E8B9D3AC99F8E4D12345678901FE9A2BC7D85F7654E28BF98BCD123456",
        },
```

new config:

```
        data={
            CONF_HOST: "192.168.2.108",
            CONF_USERNAME: "grumpypanda@china.cn",
            CONF_PASSWORD: "you-stole-my-pandas",
            CONF_USER_ID: "52C3F9E8B9D3AC99F8E4D12345678901FE9A2BC7D85F7654E28BF98BCD123456",
        },
```

Reply:
The context and meaning of the config values doesn't change, so this can just be a minor version migration where we bump the version of MINOR_VERSION to 2 and then implement a function in here which gathers the rest of the data and then sets it.